packages/editor/src/hooks/align.js

```diff
@@ -77,7 +77,17 @@ export const withToolbarControls = createHigherOrderComponent( ( BlockEdit ) =>
 	return ( props ) => {
 		const validAlignments = getBlockValidAlignments( props.name );
 
-		const updateAlignment = ( nextAlign ) => props.setAttributes( { align: nextAlign } );
+		const updateAlignment = ( nextAlign ) => {
+			if ( ! nextAlign ) {
+				const blockType = getBlockType( props.name );
+				const blockDefaultAlign = get( blockType, [ 'attributes', 'align', 'default' ] );
+				if ( blockDefaultAlign ) {
+					props.setAttributes( { align: 'none' } );
```

Comment:
Minor: We could simplify this to assign `nextAlign = 'none';` and it would fall through to same effect.